File: covilo-web-ui/src/application/core/models/location-region.ts
import { LocationCountry } from "./location-country"

/**
 *
 */
export interface LocationRegion {
  key: string
  domesticName: string
  country: LocationCountry
}

/**
 * Shortcut for the `LocationRegion` collection.
 */
export declare type LocationRegions = LocationRegion[]

File: covilo-core/src/main/kotlin/me/kvdpxne/covilo/domain/model/LocationRegion.kt
package me.kvdpxne.covilo.domain.model

import java.io.Serializable
import java.util.UUID

/**
 * Shortcut to the [LocationRegion] collection.
 */
typealias LocationRegions = Collection<LocationRegion>

/**
 * @param identifier
 * @param key
 * @param domesticName
 * @param country
 */
data class LocationRegion(
  val identifier: UUID = UUID.randomUUID(),
  val key: String,
  val domesticName: String,
  val country: LocationCountry
) : Serializable {

  companion object {
    private const val serialVersionUID: Long = 2045985117L
  }

  /**
   * Quick access to the [LocationCountry] key field.
   */
  val countryKey: String
    get() = country.key
}
